Modules/ITS/src/ITSDecodingErrorCheck.cxx

```diff
@@ -29,15 +30,41 @@ namespace o2::quality_control_modules::its
 
 Quality ITSDecodingErrorCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
 {
+  std::vector<int> vDecErrorLimits = convertToArray<int>(o2::quality_control_modules::common::getFromConfig<string>(mCustomParameters, "DecLinkErrorLimits", ""));
+  if (vDecErrorLimits.size() != o2::itsmft::GBTLinkDecodingStat::NErrorsDefined) {
+    LOG(error) << "Incorrect vector with DecodingError limits, check .json" << ENDM;
+    doFlatCheck = true;
+  }
+
   Quality result = Quality::Null;
 
   for (auto& [moName, mo] : *moMap) {
     (void)moName;
     if ((mo->getName() == "General/LinkErrorPlots") || (mo->getName() == "General/ChipErrorPlots")) {
       result = Quality::Good;
       auto* h = dynamic_cast<TH1D*>(mo->getObject());
-      if (h->GetMaximum() > 200) {
-        result.set(Quality::Bad);
+
+      if (doFlatCheck || mo->getName() == "General/ChipErrorPlots") {
+        if (h->GetMaximum() > 200) {
+          result.set(Quality::Bad);
+        }
+      } else {
+        for (int iBin = 1; iBin <= h->GetNbinsX(); iBin++) {
+
+          if (vDecErrorLimits[iBin - 1] < 0)
+            continue; // skipping bin
+
+          if (vDecErrorLimits[iBin - 1] <= h->GetBinContent(iBin)) {
+            vListErrorIdBad.push_back(iBin - 1);
+            result.set(Quality::Bad);
+            result.addReason(o2::quality_control::FlagReasonFactory::Unknown(), Form("BAD: ID = %d, %s", iBin - 1, sErrorDesc[iBin - 1].Data()));
+          } else if (vDecErrorLimits[iBin - 1] / 2 < h->GetBinContent(iBin)) {
+            vListErrorIdMedium.push_back(iBin - 1);
+            if (result != Quality::Bad)
+              result.addReason(o2::quality_control::FlagReasonFactory::Unknown(), Form("Medium: ID = %d, %s", iBin - 1, sErrorDesc[iBin - 1].Data()));
+            result.set(Quality::Medium);
```

Comment:
I would include this in the `if (result != Quality::Bad)` just above so the set quality = medium only if the quality is not already bad.

Reply:
Thanks! Indeed, it was a bug. Fixed